django/core/handlers/base.py

```diff
@@ -69,6 +75,11 @@ def load_middleware(self):
                 self._response_middleware.insert(0, mw_instance.process_response)
             if hasattr(mw_instance, 'process_exception'):
                 self._exception_middleware.insert(0, mw_instance.process_exception)
+            if callable(mw_instance):
+                self._middleware.insert(0, mw_instance)
+
+        if middleware_count == len(self._middleware):
```

Comment:
You are missing that `self._middleware` is only set if the middleware is callable, ergo a newstyle one…